tests/rsapss.cc

```diff
@@ -0,0 +1,371 @@
+                                          sgnt.data());
+
+  ASSERT_EQ(got, got_twin) << "`Hacl_RSAPSS_rsapss_sign(...)` deviates from "
+                              "`Hacl_RSAPSS_rsapss_skey_sign`.";
+  ASSERT_EQ(sgnt, sgnt_twin) << "`Hacl_RSAPSS_rsapss_sign(...)` deviates from "
+                                "`Hacl_RSAPSS_rsapss_skey_sign`.";
+
+  *out = got;
+}
+
+// TODO: Remove Copy&Paste. Is `hash_len` in `Hacl_RSAPSS.c` part of API?
+static inline uint32_t
+hash_len(Spec_Hash_Definitions_hash_alg a)
+{
+  switch (a) {
+    case Spec_Hash_Definitions_MD5: {
+      return (uint32_t)16U;
+    }
+    case Spec_Hash_Definitions_SHA1: {
+      return (uint32_t)20U;
+    }
+    case Spec_Hash_Definitions_SHA2_224: {
+      return (uint32_t)28U;
+    }
+    case Spec_Hash_Definitions_SHA2_256: {
+      return (uint32_t)32U;
+    }
+    case Spec_Hash_Definitions_SHA2_384: {
+      return (uint32_t)48U;
+    }
+    case Spec_Hash_Definitions_SHA2_512: {
+      return (uint32_t)64U;
+    }
+    case Spec_Hash_Definitions_Blake2S: {
+      return (uint32_t)32U;
+    }
+    case Spec_Hash_Definitions_Blake2B: {
+      return (uint32_t)64U;
+    }
+    default: {
+      throw std::invalid_argument(
+        "Incomplete match of Spec_Hash_Definitions_*.");
+    }
+  }
+}
+
+TEST(RsaPssSignSuite, Group)
+{
+  bytes e = from_hex("010001");
+  bytes d = from_hex(
+    "12ddcf5652e462db7bd689e1604cf27dacb7435105880c8acac24ef9302c29bc819eaee139"
+    "66d471114053e17d8ae3bca57460d1b177f8bd37bfbbee243cb5e3dde2ae34dff6b3095939"
+    "c5c74d56a674a12b270d8213a6268ec3f332dd9cf746ba097b6ce8490be4dabfb83d02560e"
+    "da766a3551681725230f31f7a67bc8f5e8968103426eac652a30893251431e434597c06487"
+    "b05b49b7a6d2e4d263de4f7e7788471b19e8aa64f3dce41bf1f55f057d50187f95379a7f93"
+    "09f6afa62ceca1e988df7c8dc484101349ca131fe9c4b4d42c63c788dc6e6ce93f11a0567c"
+    "2830022c5ee73c1c55c668e2cdcc1fb88c91bfdd33e014c29cb8af2e84c14cf9f8ad");
+  bytes n = from_hex(
+    "bb6707ae65f4ee9e65ee1a1c08b431e556cd1000dc5358b97098c9546de8ef9b5a227cbd89"
+    "fbca5fa1b0e7527cb4fd66d934f0edc166cf6f7944fb44997a0023885c319f25b7927b0c03"
+    "74132f5ab38de2bfb25228ab4cf4c3932662d4af7dab73e2da520016b7df4d97575ffb90ff"
+    "b0bda1b791f6e09cd70bc04bdc19b757279f271476fe774737ab816d0da86254a45cd98d5b"
+    "ce77ccd950e1f16f572a4e45a292b501e6394e2e49cf547302222529cb754d3d255512fe83"
+    "5874020d9f8662e22c000c8af6247be141bab2abf4a712ee1590d260bf3bb907a151604ec9"
+    "354fc806f86eeaeb4df53dd822412189567fb5fc0bc2082a7c4f613834a72f985c27");
+
+  std::vector<bytes> tests{
+    from_hex(""),
+    from_hex("00"),
+    from_hex("aa"),
+    from_hex("ff"),
+    from_hex("0000000000000000"),
+    from_hex("aaaaaaaaaaaaaaaa"),
+    from_hex("ffffffffffffffff"),
+    from_hex("000000000000000000"),
+    from_hex("aaaaaaaaaaaaaaaaaa"),
+    from_hex("ffffffffffffffffff"),
+  };
+
+  std::vector<Spec_Hash_Definitions_hash_alg> hashes{
+    // TODO(API): How to know what is supported?
```

Comment:
If you want to leave the todo in, please add an issue and link it here.

Reply:
See #100.